### PR TITLE
[Android Auto] Porting the androidauto module to the nav-sdk

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

tl;dr This is not publishing an artifact. It is copy paste from 1tap repo. It works in the `qa-test-app`. More effort needed to run tests with circleci and to create automated releases that are separate from the navigation-sdk train.

Porting the android auto solution we have working in 1tap. In order to reduce thrash on changes, this is proposing we create a v0.1.0 artifact for Android Auto _as is_. This way we can delete the solution from 1tap, examples, and create a baseline working solution.

I'm also choosing to leave out the linter in this first release, because the linter in 1tap is different. 1tap, the solution that is being tested with real drivers, remains the "source of truth" solution. This will allow us to continue to fix bugs in 1tap, and copy paste the results into `mapbox-navigation-android`. After we release a `v0.1.0`, iteration can begin within the nav-sdk repository - and this will become the new source of truth with versioned releases. 

- [x] Port over the module
- [x] Create integration in `qa-test-app`
- Hook up unit tests and instrumentation tests into ci (out of scope of this change)
- Create a v0.1.0 release (out of scope of this change)

### Screenshots or Gifs

Screen Shot 2022-04-13 at 1 17 00 PM

### CODE REVIEW

File: libnavui-androidauto/build.gradle [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    api(project(":libnavui-maneuver"))
+    api(project(":libnavui-maps"))
+    api(project(":libnavui-speedlimit"))
+    api(project(":libnavui-tripprogress"))
+    api(project(":libnavui-voice"))
+    implementation(project(":libnavui-resources"))
+    implementation(project(":libnavui-util"))
+
+    api("com.mapbox.search:mapbox-search-android:1.0.0-beta.27")
+    api("com.mapbox.search:mapbox-search-android-ui:1.0.0-beta.27")
```
Comment (Zayankovsky) [inline libnavui-androidauto/build.gradle:48]: Do we really want to depend on Search UI SDK? Looks like we only need it for some feedback drawable resources. I think we should simply copy them to Android Auto, because feedback views are likely going to be removed from Search UI SDK soon and these resources are internal anyway (IDE shows a warning).

Comment (kmadsen) [reply libnavui-androidauto/build.gradle:48]: Yeah I agree. I don't want to change it here though, either we change it in 1tap and copy-paste to update here. Or we merge as is and consider it follow up.

Transitioning the work from 1tap to nav-sdk

Comment (kmadsen) [reply libnavui-androidauto/build.gradle:48]: Ok went ahead with it, updated!

File: libnavui-androidauto/build.gradle [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+dependencies {
+    api(dependenciesList.mapboxMapsAndroidAuto)
+    api(project(":libnavui-base"))
+    api(project(":libnavui-maneuver"))
+    api(project(":libnavui-maps"))
+    api(project(":libnavui-speedlimit"))
+    api(project(":libnavui-tripprogress"))
+    api(project(":libnavui-voice"))
```
Comment (abhishek1508) [inline libnavui-androidauto/build.gradle:43]: Can be replaced with `libnavigation-android`

File: libnavui-androidauto/build.gradle [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    api("com.mapbox.search:mapbox-search-android:1.0.0-beta.29")
+
+    implementation(dependenciesList.androidXAppCompat)
+    implementation(dependenciesList.coroutinesCore)
+    implementation(dependenciesList.androidXLifecycleRuntime)
+    implementation(dependenciesList.androidXLifecycleLivedata)
+    implementation(dependenciesList.androidXConstraintLayout)
+    implementation("androidx.datastore:datastore-preferences:1.0.0")
+
+    implementation(dependenciesList.androidXAppCompat)
```
Comment (abhishek1508) [inline libnavui-androidauto/build.gradle:56]: Already added above

**Review by abhishek1508 — COMMENTED** [review]
Curious to find out why do we have `examples` hosted in this module?
@kmadsen

File: libnavui-androidauto/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+
```
[the remark this answers is not in the document]
Comment (kmadsen) [reply libnavui-androidauto/README.md:1]: Agree we should add screenshots but we're still copy-pasting

**Review by abhishek1508 — APPROVED** [review]

**Comment by kmadsen** [comment]
The test that failed is existing https://github.com/mapbox/mapbox-navigation-android/issues/5659
